### PR TITLE
Remove signature strip and ensure compression type isn't changed

### PR DESCRIPTION
**Pull request by JohanLorenzo**
This avoids https://github.com/mozilla-services/autograph/issues/164 on Firefox Focus.

Even though signature strip isn't the root cause of the bug, it is removed because the operation is too risky from a compression standpoint. APKs are too smart than regular Zip archives.

### CODE REVIEW

File: signingscript/utils.py [hunk 1]
```diff
@@ -167,3 +167,13 @@ def is_autograph_signing_format(format_):
...
+
+
+def is_apk_autograph_signing_format(format_):
+    """Return bool of whether a signing format is an APK.
+
+    Args:
+        format_ (str): the format to check
+
+    """
+    return format_ and format_ in ('autograph_fennec', 'autograph_focus')
```
Comment (MihaiTabara) [inline signingscript/utils.py]: Hm, potential dupe with https://github.com/mozilla-releng/signingscript/pull/88/files#diff-c18cb9a0544a9707c29e2a2cf9d2d23eR162 ? I suppose above refers to `mar` and others as well but just wanted to check it's intended. If we keep this, we'll have to spin up another condition for when we'll publish the `reference-brower` or potentially `android-components`. But it's definitely not blocking for now.

File: signingscript/test/integration/test_autograph.py [hunk 2]
```diff
@@ -182,37 +197,45 @@ def _assert_apk_signature(keystore_path, apk_path, certificate_alias):
...
 
     keystore_path = os.path.join(tmpdir, 'keystore')
     certificate_path = os.path.join(TEST_DATA_DIR, 'autograph_apk.pub')
-    certificate_alias = 'autograph_apk'
+    certificate_alias = 'autograph_focus'
     _instanciate_keystore(keystore_path, certificate_path, certificate_alias)
 
     await async_main(context)
 
     signed_path = os.path.join(tmpdir, 'artifact', file_name)
```
Comment (MihaiTabara) [inline signingscript/test/integration/test_autograph.py]: Unclear to me who's generating this signed bit but I'll dig after reviewing to not block this PR.

File: signingscript/test/integration/test_autograph.py [hunk 3]
```diff
@@ -182,37 +197,45 @@ def _assert_apk_signature(keystore_path, apk_path, certificate_alias):
...
-    _assert_apk_signature(keystore_path, signed_path, certificate_alias)
+    assert _verify_apk_signature(keystore_path, signed_path, certificate_alias)
+
+    zip_infos_after_signature = _extract_compress_type_per_filename(signed_path)
+    for signature_file in ('META-INF/SIGNATURE.RSA', 'META-INF/SIGNATURE.SF', 'META-INF/MANIFEST.MF'):
```
Comment (MihaiTabara) [inline signingscript/test/integration/test_autograph.py]: Nit/question: on signing-servers, these files are named differently. (e.g. `DEP` for dep key, `NIGHTLY` for nightly key, etc. For Autograph, it sounds like these are standardized around `SIGNATURE`. Should we move them elsewhere - for ease of change - either wihithin some `constants.py` or puppet or alike) if this ever changes on Autograph backend side?

File: signingscript/utils.py (hunk 1 above)
Comment (JohanLorenzo) [reply signingscript/utils.py]: Yeah, that's intended.

I agree we should define a nomenclature that avoids to list every format here. Let's do it in a follow up.

File: signingscript/test/integration/test_autograph.py (hunk 3 above)
Comment (JohanLorenzo) [reply signingscript/test/integration/test_autograph.py]: This file is about testing autograph, so no need to define anything else (for now), afaik.

File: signingscript/utils.py (hunk 1 above)
Comment (MihaiTabara) [reply signingscript/utils.py]: Sounds good.

File: signingscript/test/integration/test_autograph.py (hunk 2 above)
Comment (MihaiTabara) [reply signingscript/test/integration/test_autograph.py]: Ah, that's smooth, nice! 👍

**Pull request merged**